src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1626,18 +1623,22 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             _terminal->MultiClickSelection(terminalPosition, mode);
             selectionNeedsToBeCopied = true;
         }
-        _renderer->TriggerSelection();
-
         // this is used for mouse selection,
         // so hide the markers
-        _UpdateSelectionMarkersHandlers(*this, winrt::make<implementation::UpdateSelectionMarkersEventArgs>(true));
+        _updateSelectionUI(false);
     }
 
-    void ControlCore::_updateSelectionUI()
+    // Method Description:
+    // - Updates the renderer's representation of the selection as well as the selection marker overlay in TermControl
+    // Arguments:
+    // - tryShowMarkers: if true, show the selection marker overlay. Otherwise hide it.
+    void ControlCore::_updateSelectionUI(bool tryShowMarkers)

```

Comment:
I still don't get the boolean. Shouldn't the state of the markers be informed by what kind of selection we have? Mouse = no markers, >= keyboard = markers?

Reply:
Oh you're right! I forgot we have access to know which mode we're in! I'll add that in.